bodhi/server/consumers/masher.py

```
@@ -1042,3 +1055,230 @@ def run(self):
         else:
             self.success = True
         return out, err, returncode
+
+
+class PungiMasherThread(MasherThread):
+
+    def _get_compose_dir(self, mash_path, variant_id="Server"):
+        return os.path.join(mash_path, "compose", variant_id)
+
+    def _get_pungi_conf_path(self, config_name="fedora-modular-example.conf"):
+        if config.get("pungi_config_path"):
+            return os.path.join(config.get("pungi_config_path"), config_name)
+        else:
+            return os.path.join(
+                os.path.dirname(os.getcwd()), "bodhi", "devel", "pungi", config_name
```

We shouldn't use ```cwd``` here because it could be anything. If the intention is to form a path to this particular example config, you can use ```__file__``` to find the path to this source file, and then you can use relative paths to form the path you need.

---

Alternatively, you can just drop the ```if/else``` and raise an error if ```pungi_config_path``` is not defined in ```config```. That's what I would do if it were me. It seems unlikely that any real user would want to run with an example config.

---

https://github.com/mcurlej/bodhi/pull/7